Added preserving thead and tfoot in tables - Issue #180

simonmeadows opened the pull request:
In the columnize function we can check weather we are in a table by testing the first child of $pullOutHere
If we are in a table we can clone a copy of the <thead> and <tfoot> if they exist and store thim in the columnize function that is operating on the table.
When we get to splitting the last element we add the <thead> and <tfoot> back to the table in the content source element $pullOutHere

adamwulf commented:
Awesome - this is a great idea. Can you add a test/sample.html that shows it in action and acts as a test case for it?

simonmeadows commented:
I have added a couple of sample pages to show the behaviour.
I didn't think about nested tables initially as I wasn't going to use them but it was an interesting challenge.
I have changed the way it remembers the <thead> and <tfoot> by creating an array of all of them for every table.
It iterates through the page and adds a unique ID as a class to each table then saves an object containing the class ID, the <thead>, and the <tfoot>
Every time split() returns, it calls a function to re-add the <thead> and <tfoot> to the un-columnized content so it is ready for the next round.

adamwulf commented:
awesome thanks!
